utils/snapshot_extractor.mjs

```diff
@@ -0,0 +1,83 @@
+// @format
+import "dotenv/config";
+import { resolve, dirname } from "path";
+import { fileURLToPath } from "url";
+import { once } from "events";
+import { Worker } from "worker_threads";
+
+const __dirname = dirname(fileURLToPath(import.meta.url));
+
+export default async function snapshotExtractor2(
```

Comment:
Why is this postfixed with "2"?

Reply:
Sorry, my bad. I simultaneously created two functions with different approaches and I was testing them back and forth to determine the better one. One function was named snapshotExtractor2 and the other snapshotExtractor. Forgot to remove the 2 before committing.

I will fix this.

Reply:
ok thx